conn/physic/units.go

```diff
@@ -733,3 +734,307 @@ func picoAsString(v int64) string {
 	}
 	return sign + strconv.Itoa(base) + "." + prefixZeros(3, frac) + unit
 }
+
+// Decimal is the exact representation of decimal number.
+type decimal struct {
+	// digits hold the string representation of the significant decimal digits.
+	digits string
+	// exponent is the left or right decimal shift. (powers of ten).
+	exp int
+	// neg it true if the number is negative.
+	neg bool
+}
+
+// Positive powers of 10 in the form such that powerOF10[index] = 10^index.
+var powerOf10 = [...]uint64{
+	1,
+	10,
+	100,
+	1000,
+	10000,
+	100000,
+	1000000,
+	10000000,
+	100000000,
+	1000000000,
+	10000000000,
+	100000000000,
+	1000000000000,
+	10000000000000,
+	100000000000000,
+	1000000000000000,
+	10000000000000000,
+	100000000000000000,
+	1000000000000000000,
+}
+
+// Maximum value for a int64.
+const maxInt64 = (1<<63 - 1)
+
+var maxUint64Str = "9223372036854775807"
+
+// Converts from decimal to int64, using the decimal.digits character values and
+// converting to a intermediate unit64.
+// Scale is combined with the decimal exponent to maximise the resolution and is
+// in powers of ten.
```

Comment:
We should eventually add a comment that "it tries to go as far as it can until it finds a character that cannot be parsed as a number".